Mathlib/SetTheory/Cardinal/Arithmetic.lean

```diff
@@ -817,6 +817,46 @@ end extend
 
 end Cardinal
 
+/-! ### Cardinality of ordinals -/
+
+namespace Ordinal
+
+theorem lift_card_iSup_le_sum_card {ι : Type u} [Small.{v} ι] (f : ι → Ordinal.{v}) :
+    Cardinal.lift.{u} (⨆ i, f i).card ≤ Cardinal.sum fun i ↦ (f i).card := by
+  simp_rw [← mk_toType]
+  rw [← mk_sigma, ← Cardinal.lift_id'.{v} #(Σ _, _), ← Cardinal.lift_umax.{v, u}]
+  apply lift_mk_le_lift_mk_of_injective (f := fun x ↦
+    let a := (enumIsoToType _).symm x
+    have H := (Ordinal.lt_iSup (f := f) (a := a.1)).1 a.2
+    let b := Classical.choose H
+    ⟨b, enumIsoToType (f b) ⟨a.1, Classical.choose_spec H⟩⟩)
+  intro a b h
+  rw [Sigma.mk.inj_iff] at h
+  suffices ∀ {x y z a b hx hy}, x = y → HEq (enumIsoToType x ⟨↑((enumIsoToType z).symm a), hx⟩)
+    (enumIsoToType y ⟨↑((enumIsoToType z).symm b), hy⟩) → a = b from this (congr_arg f h.1) h.2
+  intro _ _ _ _ _ _ _ h
+  subst h
+  simp [← Subtype.eq_iff]
```

Comment:
Why not use lift_mk_le_lift_mk_of_**surjective** instead?
```suggestion
  simp_rw [← mk_toType]
  rw [← mk_sigma, ← Cardinal.lift_id'.{v} #(Σ _, _), ← Cardinal.lift_umax.{v, u}]
  apply lift_mk_le_lift_mk_of_surjective (f := enumIsoToType _ ∘ (⟨(enumIsoToType _).symm ·.2,
    lt_of_lt_of_le (mem_Iio.mp ((enumIsoToType _).symm _).2) (Ordinal.le_iSup _ _)⟩))
  rw [EquivLike.comp_surjective]
  intro ⟨x, hx⟩
  have ⟨i, hi⟩ := Ordinal.lt_iSup.mp hx
  exact ⟨⟨i, enumIsoToType _ ⟨x, hi⟩⟩, by simp⟩
```

Reply:
Nice! This is much cleaner!